cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -493,6 +493,14 @@ endif()
 
 #---Check for GSL library---------------------------------------------------------------
 if(mathmore OR builtin_gsl)
+  if(builtin_gsl AND NO_CONNECTION)
```

Comment:
humm ... we might be missing support for the following scenario:

* download and configure (and possibly build once) on a node that has network access 
* re-configure on the same build directory but from a node that had no network access.

This scenario *should* be supported as it is a means to download and build on a non-network accessible node.

I.e. we might also need to check whether the buitin has *already* been download before failing.

Reply:
Thanks @pcanal for the nice suggestion. i also thought that our CMake system was not complex enough... I'll try to come with something. I already spent several days with the dependencies hell, I can spend several more to try all possible combinations 😉

Reply:
@pcanal is there documentation somewhere that would explain how to do what you talk about? I may be interested by such option in some cases.
PS: I can start a message on the ROOT forum not to pollute Github with this discussion.

Reply:
I think it's fine to not support this until there is demand, and keep the build system simpler. If people need to build without network we can always ask them to not use builtins.

Reply:
> is there documentation somewhere that would explain how to do what you talk about? I 

Not really.  It is just: 'on machine that has access to the internet on the disk you need to build on, do a git clone and then a configuration.  In some cases that is sufficient and you can continue the build on a machine that has no access to the internet, sometimes you need to do a first (successully) build on the machine with internet access.